### PR TITLE
Plugin Property types improvements

### PR DESCRIPTION
**Pull request by pauloouriques**
## Change description

This pull request contains multiple property types improvements. 

### Development

- [x] Application changes have been tested appropriately

### Impact

- [x] Code follows company security practices and guidelines
- [x] Security impact of change has been considered
- [x] Performance impact of change has been considered
- [x] Possible migration needs considered (model migrations, config migrations, etc.)

Please explain any security, performance, migration, or other impacts if relevant:

### Code review

- [x] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached where applicable.
- [x] Relevant tags have been added to the PR (bug, enhancement, internal, etc.)

### CODE REVIEW

File: plugins/@grouparoo/salesforce/src/lib/export/mapping.ts [hunk 1]
```diff
@@ -86,22 +86,22 @@ const supportedTypeMap = {
   id: "string",
   picklist: "string",
   currency: "float",
+  masterrecord: "string", // When records are merged, the ID of the record that is saved
+  reference: "string", // Cross-references to a different object. Analogous to a foreign key field in SQL.
+  base64: "string", // Base 64-encoded binary data.
   //TODO:
-  address: null,
+  address: null, //A compound data type that contains address field data.
   date: null,
```
Comment (bleonard) [inline plugins/@grouparoo/salesforce/src/lib/export/mapping.ts]: Dates are probably the most important ones to fix on all these plugins, especially in Salesforce. For example, `CloseDate` is a required field when making an `Opportunity`

File: plugins/@grouparoo/braze/__tests__/export/export-records.ts [hunk 2]
```diff
@@ -212,6 +212,7 @@ describe("braze/exportRecords", () => {
               external_id: email1,
               first_name: "John",
               last_name: "Doe",
+              dob: new Date("1980-12-21"),
```
Comment (bleonard) [inline plugins/@grouparoo/braze/__tests__/export/export-records.ts]: This probably on'y actually be a `Date` object. So let's just test with `"1980-12-21"` to make sure it works.

File: plugins/@grouparoo/braze/src/lib/export/destinationMappingOptions.ts [hunk 3]
```diff
@@ -22,9 +22,9 @@ export const destinationMappingOptions: DestinationMappingOptionsMethod =
           { key: "phone", type: "phoneNumber", important: false },
           { key: "gender", type: "string", important: false },
           { key: "country", type: "string", important: false },
-          { key: "date_of_first_session", type: "string", important: true },
-          { key: "date_of_last_session", type: "string", important: true },
-          { key: "dob", type: "string", important: true },
+          { key: "date_of_first_session", type: "date", important: true },
```
Comment (bleonard) [inline plugins/@grouparoo/braze/src/lib/export/destinationMappingOptions.ts]: These should actually stays a strings. Current `"date"` actually means `new Date()` which is really a `"datetime"`

File: plugins/@grouparoo/braze/__tests__/export/export-records.ts (hunk 2 above)
Comment (bleonard) [reply plugins/@grouparoo/braze/__tests__/export/export-records.ts]: same thing for all the tests

File: plugins/@grouparoo/mailchimp/src/lib/shared/getMergeVars.ts [hunk 4]
```diff
@@ -49,6 +49,10 @@ function toGrouparooDestinationType(field: MailchimpMergeField) {
       return { type: "string", important: true };
     case "address":
       return { type: "string", important: true };
+    case "date":
+      return { type: "date", important: true };
```
Comment (bleonard) [inline plugins/@grouparoo/mailchimp/src/lib/shared/getMergeVars.ts]: `"string"` here

File: plugins/@grouparoo/marketo/src/lib/export/destinationMappingOptions.ts [hunk 5]
```diff
@@ -38,19 +38,19 @@ const mapTypesFromMarketoToGrouparoo = (marketoType) => {
     integer: "integer",
     boolean: "boolean",
     email: "email",
+    phone: "phoneNumber",
     float: "float",
     datetime: "date",
+    date: "date",
```
Comment (bleonard) [inline plugins/@grouparoo/marketo/src/lib/export/destinationMappingOptions.ts]: `"string"` here

Comment (bleonard) [reply plugins/@grouparoo/marketo/src/lib/export/destinationMappingOptions.ts]: And all the other ones that were null before.

**Review by bleonard — COMMENTED** [review]
Misunderstanding about date vs. datetime concepts. trying to clarify.

**Comment by evantahler** [comment]
@pauloouriques - for compliance, we really do need everyone to fill out the PR template located [here](https://github.com/grouparoo/grouparoo/blob/7ec8ce221b9f894af38935aba009dcd32a9e1691/.github/pull_request_template.md).  Please copy this into your PR description and fill it out    It's also nice to have a good description in the PR to remember why we made a certain change tied to a commit.

File: plugins/@grouparoo/mailchimp/src/lib/shared/getMergeVars.ts [hunk 6]
```diff
@@ -49,6 +49,10 @@ function toGrouparooDestinationType(field: MailchimpMergeField) {
       return { type: "string", important: true };
     case "address":
       return { type: "string", important: true };
+    case "date":
+      return { type: "string", important: true };
+    case "birthday":
+      return { type: "date", important: true };
```
Comment (evantahler) [inline plugins/@grouparoo/mailchimp/src/lib/shared/getMergeVars.ts]: 👍 these should be strings for now

File: plugins/@grouparoo/marketo/src/lib/export/destinationMappingOptions.ts [hunk 7]
```diff
@@ -38,19 +38,19 @@ const mapTypesFromMarketoToGrouparoo = (marketoType) => {
...
     boolean: "boolean",
     email: "email",
+    phone: "phoneNumber",
     float: "float",
     datetime: "date",
+    date: "string",
     score: "integer",
     currency: "float",
+    url: "url",
+    percent: "integer",
```
Comment (evantahler) [inline plugins/@grouparoo/marketo/src/lib/export/destinationMappingOptions.ts:48]: This one's interesting - I would have suggested a `float`, so then `0.6` could be sent too.  Does Marketo want whole numbers or decimals?

File: plugins/@grouparoo/salesforce/src/lib/export/mapping.ts [hunk 8]
```diff
@@ -76,7 +76,7 @@ const supportedTypeMap = {
   textarea: "string",
   datetime: "date",
   boolean: "boolean",
-  phone: "phone",
```
Comment (evantahler) [inline plugins/@grouparoo/salesforce/src/lib/export/mapping.ts:79]: oh no! This probably prevented phone numbers from being sent

File: plugins/@grouparoo/salesforce/src/lib/export/mapping.ts [hunk 9]
```diff
@@ -86,22 +86,22 @@ const supportedTypeMap = {
...
   picklist: "string",
   currency: "float",
+  masterrecord: "string", // When records are merged, the ID of the record that is saved
+  reference: "string", // Cross-references to a different object. Analogous to a foreign key field in SQL.
+  base64: "string", // Base 64-encoded binary data.
+  date: "string",
   //TODO:
-  address: null,
-  date: null,
+  address: null, //A compound data type that contains address field data.
```
Comment (evantahler) [inline plugins/@grouparoo/salesforce/src/lib/export/mapping.ts:94]: Is there anything we can do to export this better?  For example, what if was a grouparoo string, and upon export we sent something like `{street_address: ${value}}` - would that work?

**Review by evantahler — CHANGES_REQUESTED** [review]

File: plugins/@grouparoo/marketo/src/lib/export/destinationMappingOptions.ts (hunk 7 above)
Comment (pauloouriques) [reply plugins/@grouparoo/marketo/src/lib/export/destinationMappingOptions.ts:48]: only decimals as integers, look: "A percentage expressed as an integer". In other words, only 2 decimals places (0.01 = 1)

File: plugins/@grouparoo/salesforce/src/lib/export/mapping.ts (hunk 9 above)
Comment (pauloouriques) [reply plugins/@grouparoo/salesforce/src/lib/export/mapping.ts:94]: hmmm, good idea. I'll try this out.

Comment (pauloouriques) [reply plugins/@grouparoo/salesforce/src/lib/export/mapping.ts:94]: It looks like there is no way to create a new Address field, and in contacts, we don't receive it directly we receive it spread like this:  
`{...
MailingStreet: null,
      MailingCity: null,
      MailingState: null,
      MailingPostalCode: null,
      MailingCountry: null,
      MailingLatitude: null,
      MailingLongitude: null,
      MailingGeocodeAccuracy: null,
      MailingAddress: null,
      Phone: null,
      Fax: null,
      MobilePhone: null,
      ...
}`

Comment (evantahler) [reply plugins/@grouparoo/salesforce/src/lib/export/mapping.ts:94]: Dang. Ok then, thanks for checking!

**Review by evantahler — APPROVED** [review]

**Pull request merged**